### PR TITLE
GetIndexAsync should return null if index_not_found

### LINKED ISSUE #386: Expected GetIndexAsync to return null when not found, actually throws an exception
**Description**

In `MeilisearchClient` the `GetIndexAsync` method throws an exception of type `MeilisearchApiError` when the index is not found.

**Expected behavior**

The documentation written specifies that `null` will be returned in case of a non existing index.

**Screenshots or Logs**

Line 168

```
/// <summary>
/// Gets an index with the unique ID.
/// </summary>
/// <param name="uid">Unique identifier of the index.</param>
/// <param name="cancellationToken">The cancellation token for this call.</param>
/// <returns>Returns Index or Null if the index does not exist.</returns>
public async Task<Index> GetIndexAsync(string uid, CancellationToken cancellationToken = default)
{
   return await Index(uid).FetchInfoAsync(cancellationToken).ConfigureAwait(false);
}
```

**Environment (please complete the following information):**
- OS: docker pull
- Meilisearch version: v.0.30.0
- meilisearch-dotnet version: v0.13.0

### PR DESCRIPTION
**Pull request by ahmednfwela**
# Pull Request

## Related issue
Fixes #386

## What does this PR do?
- Catches `MeilisearchApiError` in `GetIndexAsync` and returns null if code is index_not_found

## PR checklist
Please check if your PR fulfills the following requirements:
- [x] Does this PR fix an existing issue, or have you listed the changes applied in the PR description (and why they are needed)?
- [x] Have you read the contributing guidelines?
- [x] Have you made sure that the title is accurate and descriptive of the changes?

Thank you so much for contributing to Meilisearch!

### CODE REVIEW

**Review by alallema — COMMENTED** [review]
My bad, I should have explained this issue better. The `GetIndexAsync` method should not return `null` in case the index is not found, it must return an error, it is the documentation that is wrong and must be changed.
Sorry

**Comment by ahmednfwela** [comment]
changed to update the documentation 👍

File: src/Meilisearch/MeilisearchClient.cs [hunk 1]
```diff
@@ -165,7 +165,8 @@ public async Task<ResourceResults<IEnumerable<Index>>> GetAllIndexesAsync(Indexe
         /// </summary>
         /// <param name="uid">Unique identifier of the index.</param>
         /// <param name="cancellationToken">The cancellation token for this call.</param>
-        /// <returns>Returns Index or Null if the index does not exist.</returns>
+        /// <returns>Returns Index or Throws if the index does not exist.</returns>
+        /// <exception cref="MeilisearchApiError">If the index doesn't exist</exception>
```
Comment (alallema) [inline src/Meilisearch/MeilisearchClient.cs]:
```suggestion
        /// <returns>Returns Index.</returns>
        /// <exception cref="MeilisearchApiError">Throws if the index doesn't exist.</exception>
```

**Review by alallema — DISMISSED** [review]
Thanks @ahmednfwela, for this PR!
Just a little suggestion

**Review by alallema — APPROVED** [review]
LGTM! 🔥

**Comment by alallema** [comment]
bors merge

**Pull request merged**